sormas-backend/src/main/java/de/symeda/sormas/backend/feature/FeatureConfigurationFacadeEjb.java

```diff
@@ -254,6 +257,35 @@ public boolean isFeatureDisabled(FeatureType featureType) {
 		return em.createQuery(cq).getSingleResult() > 0;
 	}
 
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean isPropertyValue(FeatureType featureType, FeatureTypeProperty property, boolean expectedPropertyValue) {
+
+		if (!Arrays.asList(featureType.getSupportedProperties()).contains(property)) {
+			throw new IllegalArgumentException("Feature type " + featureType + " does not support property " + property + ".");
+		}
+
+		if (!Boolean.class.isAssignableFrom(property.getReturnType())) {
+			throw new IllegalArgumentException(
+				"Feature type property " + property + " does not have specified return type " + Boolean.class.getSimpleName() + ".");
+		}
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Object> cq = cb.createQuery(Object.class);
+		Root<FeatureConfiguration> root = cq.from(FeatureConfiguration.class);
+
+		cq.where(cb.and(cb.equal(root.get(FeatureConfiguration.FEATURE_TYPE), featureType)));
+		cq.select(root.get(FeatureConfiguration.PROPERTIES));
+
+		Map<FeatureTypeProperty, Object> properties = (Map<FeatureTypeProperty, Object>) em.createQuery(cq).getSingleResult();
+
+		if (properties != null && properties.containsKey(property)) {
+			return (boolean) properties.get(property) == expectedPropertyValue;
```

Comment:
Properly sanitize boolean value, check `ConfigFacadeEjb.getBoolean`. It also shows how to deal with the `expectedPropertyValue`.

Reply:
I'm not exactly sure how you want the code to look like; something like `Boolean.parseBoolean` isn't possible because we're working with an Object, not a String. Catching a ClassCastException here and doing something else with it also isn't a good idea in my opinion because we should rather let the error happen here so that users are aware that something's wrong. Can you add a specific code example here please?

Reply:
I assume that casting whatever comes from the JSON to boolean (`null` will result in a NPE!) is probably okay then.